core/src/main/java/org/apache/calcite/rel/rules/CoreRules.java

```diff
@@ -279,6 +279,10 @@ private CoreRules() {}
   public static final FilterProjectTransposeRule FILTER_PROJECT_TRANSPOSE =
       FilterProjectTransposeRule.Config.DEFAULT.toRule();
 
+  /** Rule that pushes a {@link Filter} past a {@link Sample}. */
+  public static final FilterSampleTransposeRule FILTER_SAMPLE_TRANSPOSE =
```

Comment:
How about adding it to BASE_RUELS too? We already have many other transpose rules there.

Reply:
Hi @libenchao ,thanks for your reviewing.

It makes sense to me,I will take it.